### PR TITLE
Fix typescriptimports

### PR DESCRIPTION
**Pull request by sergioferreiradt**
Changed the imports with typemap to %pragma. Fixed the order of handler execution by querying the parse tree to find the tsimports pragma just after starting top()

### CODE REVIEW

File: Examples/test-suite/typescript_import.i [hunk 1]
```diff
@@ -19,3 +13,7 @@ class C {
 };
 %}
 
+%pragma(javascript) tsimports=%{
+import { D } from './typescript-import-d-types';
+%}
+
```
Comment (vadz) [inline Examples/test-suite/typescript_import.i]: It doesn't really matter, of course, but I'd keep this pragma above, it makes more sense to order things in the same order in the `.i` file in which they appear in the output, IMHO.

**Review by vadz — COMMENTED** [review]
This is not necessarily the approach I'd use because I think it doesn't behave correctly if you get 2 (or more) such pragmas: normally you'd expect the latter one to override the former, but with your approach I think only the first one is taken into account and all the other ones are completely ignored.

I.e. I'd delay the generation of the output file until the end, i.e. when you already have all the pieces. Would it really be that much more difficult to do?

File: Examples/test-suite/typescript_import.i (hunk 1 above)
Comment (sergioferreiradt) [reply Examples/test-suite/typescript_import.i]: No. It is not that difficult to do. I need to collect all the pieces into a collection of interfaces/enums and one string for the pragma and in the end write them. Or maybe do it with two string (one for the interfaces/enums and another one for the pragma) to keep it simple.

Comment (vadz) [reply Examples/test-suite/typescript_import.i]: Yes, I'd just use 2 strings (and add more if more are needed later, of course).

File: Source/Modules/javascript.cxx [hunk 2]
```diff
@@ -2725,24 +2738,42 @@ int TypeScriptTypes::classHandler(Node *n)
...
+/**
+ * Handler executed when a pragma is found in the Parse Tree
+ *
+ * @param n The node found that have the type %pragma
+ */
+int TypeScriptTypes::pragmaDirective(Node *n) {
+  if (generateTsTypes)
+  {
+    String *tsImportsLanguage = NewString("javascript");
+    String *tsImportsName = NewString("tsimports");
```
Comment (vadz) [inline Source/Modules/javascript.cxx]: Very minor, but I don't think you need `String`s for this, AFAIR `Strcmp()` works with `char*` strings directly too.

**Review by vadz — APPROVED** [review]
Looks good to me, thanks!

**Review by vadz — APPROVED** [review]
Thanks!

**Pull request merged**